app/controllers/tickets_controller.rb

```diff
@@ -164,76 +168,80 @@ def create
         end
       end
     else
+      using_hook = false
       @ticket = Ticket.new(ticket_params)
-      if current_user.nil? && !verify_recaptcha
-        return render 'new'
-      end
     end
 
-    send_notification_email
-
-    respond_to do |format|
-      format.html do
-
-        respond_to_html
-
+    if ::Rails.application.config.ticket_is_open_to_the_world  == false && current_user.nil? && using_hook == false
+      render :status => :forbidden, :text => "Access Denied"
+    elsif can_create_a_ticket(using_hook) && save_current_ticket
+      notify_incoming @ticket
+      
+      respond_to do |format|
+        format.json { render json: @ticket, status: :created }
+        format.html {
+          if current_user.nil?
+            flash.now[:notice] = I18n::translate(:ticket_added_public)
+            render 'create'
+          else
+            redirect_to ticket_url(@ticket), notice: I18n::translate(:ticket_added)
+          end
+        }
       end
-
-      format.json do
-
-        respond_to_json
-
+    else
+      respond_to do |format|
+        format.html {
+          @email_addresses = EmailAddress.verified.ordered
+          render 'new'
+        }
+        format.json {
+          if @ticket.nil?
+            render json: {}, status: :created  # bounce mail handled correctly
+          else
+            render json: @ticket, status: :unprocessable_entity
+          end 
+        }
       end
-
-      format.js { render }
     end
   end
 
   protected
-  def respond_to_json
-    if @ticket.nil?
-      render json: {}, status: :created  # bounce mail handled correctly
-    elsif @ticket.valid?
-      render json: @ticket, status: :created
+  def can_create_a_ticket(using_hook)
+    if @ticket.nil? || !@ticket.valid?
+      flash.now[:alert] = I18n::translate(:form_validation_error)
```

Comment:
On which redirect do you see this message appear? As I was seeing it, the most likely place it should appear is [here](https://github.com/mickael-kerjean/brimir/blob/2e0616e1f71d662d93e4a5d1b868e7ca1674221e/app/controllers/tickets_controller.rb#L195)

In this function, I just tried to separate the authorization logic, from the rest of the controller. 
That way the rest of the controller doesn't have to know about the logic for generating alert messages. 
I believed this approach was cleaner and easier to expand than pushing everything inside the  create method